### ISSUE TITLE
Renovate failing to make PR on Gitea 1.16.1

### ISSUE THREAD

**Issue by redstonedesigner** [renovatebot/renovate#14059]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

31.68.0

### Please select which platform you are using if self-hosting.

Gitea

### If you're self-hosting Renovate, tell us what version of the platform you run.

1.16.1

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

When attempting to post a Pull Request to a repository, there is a 422 Unprocessable Entity returned by Gitea to Renovate.  This seems to be due to a malformed JSON input from Renovate to Gitea.

### Relevant debug logs

<details><summary>Logs</summary>

```
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 50,
           "logContext": "21H0P7t9qc2ATU2w-6qhR",
           "repository": "Owner/Repository",
           "err": {
             "name": "HTTPError",
             "code": "ERR_NON_2XX_3XX_RESPONSE",
             "timings": {
               "start": 1644236061632,
               "socket": 1644236061633,
               "lookup": 1644236061634,
               "connect": 1644236061644,
               "secureConnect": 1644236061657,
               "upload": 1644236061658,
               "response": 1644236061745,
               "end": 1644236061747,
               "phases": {
                 "wait": 1,
                 "dns": 1,
                 "tcp": 10,
                 "tls": 13,
                 "request": 1,
                 "firstByte": 87,
                 "download": 2,
                 "total": 115
               }
             },
             "message": "Response code 422 (Unprocessable Entity)",
             "stack": "HTTPError: Response code 422 (Unprocessable Entity)\n    at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:117:42)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
             "options": {
               "headers": {
                 "user-agent": "RenovateBot/31.68.0 (https://github.com/renovatebot/renovate)",
                 "authorization": "***********",
                 "accept": "application/json",
                 "content-type": "application/json",
                 "content-length": "1515",
                 "accept-encoding": "gzip, deflate, br"
               },
               "url": "https://git.example.com/api/v1/repos/Owner/Repo/pulls",
               "hostType": "gitea",
               "username": "",
               "password": "",
               "method": "POST",
               "http2": false
             },
             "response": {
               "statusCode": 422,
               "statusMessage": "Unprocessable Entity",
               "body": {
                 "message": "[]: json: string unexpected end of JSON input",
                 "url": "https://git.example.com/api/swagger"
               },
               "headers": {
                 "date": "Mon, 07 Feb 2022 12:14:21 GMT",
                 "content-type": "application/json;charset=utf-8",
                 "content-length": "109",
                 "connection": "close",
                 "set-cookie": [
                   "i_like_gitea=a6aee114e463ea56; Path=/; HttpOnly; SameSite=Lax",
                   "_csrf=xy9zokjrrytCNjK6gAvriPYbfGo6MTY0NDIzNjA2MTY4MTY4ODQ1OQ; Path=/; Expires=Tue, 08 Feb 2022 12:14:21 GMT; HttpOnly; SameSite=Lax"
                 ],
                 "x-content-type-options": "nosniff",
                 "x-frame-options": "SAMEORIGIN",
                 "cf-cache-status": "DYNAMIC",
                 "expect-ct": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\"",
                 "report-to": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=U0M5HgqXrYP9fFuumd%2B%2BSbI6%2Feu7ssvDxOfJTDEWheTWi%2FtBvHx8PrvJjNdZ1HScXNZ4HiIfQnX2kIPLwwY2n1kz3gOOGU8s3VaMpIuF%2FldoiStJFggLax7CSIz1JtI8paoq%2FoXDlw%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}",
                 "nel": "{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}",
                 "server": "cloudflare",
                 "cf-ray": "6d9c8999680d639e-ORD",
                 "alt-svc": "h3=\":443\"; ma=86400, h3-29=\":443\"; ma=86400"
               },
               "httpVersion": "1.1"
             }
           },
           "msg": "Repository has unknown error"
         }```

</details>


### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
Renovate works fine with many other instances of Gitea, so unfortunately you will need to do some reproducing/troubleshooting if we are to understand what's going wrong with yours.

**Comment by rarkins** [comment]
When you say:

> It used to work, and then stopped

What changed? Gitea version, or Renovate version?

**Comment by rarkins** [comment]
In this case we don't know if it's a Gitea change or Renovate change so need to wait for further feedback

**Comment by viceice** [comment]
I don't see any issues on my gitea instance

**Comment by redstonedesigner** [comment]
Appears to be an upstream issue with Gtea rather than issue with Renovate.

The permissions system was changed in 1.16.0 leading to the wrong value "read-only" being returned by Renovate's permission checks, rather than "read-write".  Currently, the only way to fix this is to add the Renovate user to a team with "Administrator Access" to repositories.

**Comment by xoxys** [comment]
I'm facing the same issue. @redstonedesigner is there an existing Gitea issue already?

**Comment by xoxys** [comment]
The strange thing is, it is still working on other repos on the same Gitea (v1.16.1) server. So the current operation is:

- successful Renovate PR creation on multiple repos
- faulty PR creation on a single repo with the same error message as in the initial issue; what looks strange to me is that the branch + commit to the branch was created and pushed successfully so "just" the PR creation fails

The described workaround does not fix the issue for me, and due to the fact that everything works on other repos this does not look like a gitea or permission issue to me.

@viceice @rarkins Would you mind reopening the issue, or should I create a new one? Thanks for your help.

**Comment by redstonedesigner** [comment]
> I'm facing the same issue. @redstonedesigner is there an existing Gitea issue already?

I don't believe there is currently an open issue for this on Gitea.

**Comment by viceice** [comment]
Open a new bug report with an exact reproduction. So it's probably a PR body issue.
Maybe you can reproduce it on https://try.gitea.io

**Comment by viceice** [comment]
this is now fixed upstream in v1.20.0
